draft-ietf-tls-tls13.md

```diff
@@ -2323,102 +2315,138 @@ The "extension_data" field of this extension contains a
+  MGF1. The digest used in the mask generation function and the digest
+  being signed are both the corresponding hash algorithm as defined in
+  {{SHS}}. When used in signed TLS handshake messages (see
+  {{digital-signing}}), the length of the salt MUST be equal to the length
+  of the digest output.
+
+ed25519, ed448
+: Indicates a signature algorithm using EdDSA as defined in
+  {{I-D.irtf-cfrg-eddsa}} or its successors. Note that these correspond to
+  the "PureEdDSA" algorithms and not the "prehash" variants.
 {:br }
 
 The semantics of this extension are somewhat complicated because the cipher
-suite indicates permissible signature algorithms but not hash algorithms.
-{{server-certificate-selection}} and {{key-share}} describe the
-appropriate rules.
+suite adds additional constraints on signature algorithms.
+{{server-certificate-selection}} describes the appropriate rules.
 
-Clients offering support for SHA-1 for backwards compatibility MUST do so by listing
-those hash/signature pairs as the lowest priority (listed after all other
-pairs in the supported_signature_algorithms vector). TLS 1.3 servers MUST NOT
-offer a SHA-1 signed certificate unless no valid certificate chain can be
-produced without it (see {{server-certificate-selection}}).
+rsa_pkcs1_sha1 and dsa_sha1 SHOULD NOT be offered. Clients offering these
+values for backwards compatibility MUST list them as the lowest priority
+(listed after all other algorithms in the supported_signature_algorithms
+vector). TLS 1.3 servers MUST NOT offer a SHA-1 signed certificate unless no
+valid certificate chain can be produced without it (see
+{{server-certificate-selection}}).
 
 The signatures on certificates that are self-signed or certificates that are
 trust anchors are not validated since they begin a certification path (see
 {{RFC5280}}, Section 3.2).  A certificate that begins a certification
-path MAY use a hash or signature algorithm that is not advertised as being
-supported in the "signature_algorithms" extension.
+path MAY use a signature algorithm that is not advertised as being supported
+in the "signature_algorithms" extension.
+
+Note that TLS 1.2 defines this extension differently. TLS 1.3 implementations
+willing to negotiate TLS 1.2 MUST behave in accordance with the requirements of
+{{RFC5246}} when negotiating that version. In particular:
 
-Note: TLS 1.3 servers might receive TLS 1.2 ClientHellos which do not contain
-this extension. If those servers are willing to negotiate TLS 1.2, they MUST
-behave in accordance with the requirements of {{RFC5246}} when negotiating
-that version.
+* TLS 1.2 ClientHellos may omit this extension.
+
+* In TLS 1.2, the extension contained hash/signature pairs. The pairs are
+  encoded in two octets, so SignatureAlgorithm values have been allocated to
+  align with TLS 1.2's encoding. Unallocated pairs are deprecated as of TLS
+  1.3. These values MUST NOT be offered or negotiated by any implementation.
```

Comment:
It occurs to I don't actually know that all existing TLS 1.2 ECDSA servers are willing to sign SHA-256 or better.  This probably wants some SHOULD NOT'd obsolete_ecdsa_sha1(0x0103) allocation that's ignored in TLS 1.3 but might be advertised for TLS 1.2, at least until we have some evidence that this is not needed for TLS 1.2 compatibility.

(Whether ECDSA-SHA-1 certificates exist in the Web PKI is largely answerable with CT, but my initial explorations suggested that some servers also insist on signing ServerKeyExchange with SHA-1. Almost all RSA of course, but I didn't think to separate Chrome's current metrics by hash type.)